src/box/lua/config/descriptions.lua

```diff
@@ -0,0 +1,2446 @@
+    actual value, use `index_object:stat().range_size`.
+]])
+
+M['vinyl.read_threads'] = format_text([[
+    The maximum number of read threads that vinyl can use for concurrent
+    operations, such as I/O and compression.
+]])
+
+M['vinyl.run_count_per_level'] = format_text([[
+    The maximum number of runs per level in the vinyl LSM tree. If this
+    number is exceeded, a new level is created. The `vinyl.run_count_per_level`
+    setting is a default value for the run_count_per_level option passed to
+    `space_object:create_index()`.
+]])
+
+M['vinyl.run_size_ratio'] = format_text([[
+    The ratio between the sizes of different levels in the LSM tree.
+    The `vinyl.run_size_ratio` setting is a default value for the
+    run_size_ratio option passed to `space_object:create_index()`.
+]])
+
+M['vinyl.timeout'] = format_text([[
+    The vinyl storage engine has a scheduler that performs compaction.
+    When vinyl is low on available memory, the compaction scheduler may
+    be unable to keep up with incoming update requests. In that situation,
+    queries may time out after vinyl.timeout seconds. This should rarely occur,
+    since normally vinyl throttles inserts when it is running low on compaction
+    bandwidth. Compaction can also be initiated manually with
+    `index_object:compact()`.
+]])
+
+M['vinyl.write_threads'] = format_text([[
+    The maximum number of write threads that vinyl can use for some
+    concurrent operations, such as I/O and compression.
+]])
+
+-- }}} vinyl configuration
+
+-- {{{ wal configuration
+
+M['wal'] = format_text([[
+    This section defines configuration parameters related to write-ahead log.
+]])
+
+M['wal.cleanup_delay'] = format_text([[
+    The delay in seconds used to prevent the Tarantool garbage collector from
+    immediately removing write-ahead log files after a node restart. This
+    delay eliminates possible erroneous situations when the master deletes
+    WALs needed by replicas after restart. As a consequence, replicas sync
+    with the master faster after its restart and don't need to download all
+    the data again. Once all the nodes in the replica set are up and
+    running, a scheduled garbage collection is started again even if
+    `wal.cleanup_delay` has not expired.
+
+]])
+
+M['wal.dir'] = format_text([[
+    A directory where write-ahead log (`.xlog`) files are stored. A relative
+    path in this option is interpreted as relative to `process.work_dir`.
+
+    By default, WAL files and snapshots are stored in the same directory.
+    However, you can set different values for the `wal.dir` and `snapshot.dir`
+    options to store them on different physical disks for performance matters.
+]])
+
+M['wal.dir_rescan_delay'] = format_text([[
+    The time interval in seconds between periodic scans of the write-ahead-log
+    file directory, when checking for changes to write-ahead-log files for the
+    sake of replication or hot standby.
+]])
```

Comment:
To be honest, this description in unclear for me.

I understand that we need to scan the WAL directory from time to time and that it is somehow related to replication and hot standby.

But what changes if we scan the directory frequently or rarely? Which actions are performed based on the scan result?

Feel free to refuse to collect the details. We can work on the description later. I've noted it in case you're interested to look into it yourself.

Reply:
I tried searching, but I haven't come up with anything clear yet. Let's return here back later.

Reply:
OK.